code/game/objects/items/weapons/blades.dm

```diff
@@ -344,39 +348,67 @@
 
 /obj/item/stack/throwing_knife/equipped(mob/user, slot)
 	. = ..()
-	if(!isliving(user))
+	if(user.get_active_held_item() != src && user.get_inactive_held_item() != src)
 		return
-	var/mob/living/living_user = user
-	if(living_user.get_active_held_item() != src && living_user.get_inactive_held_item() != src)
-		return
-	RegisterSignal(user, COMSIG_MOB_ITEM_AFTERATTACK, .proc/throw_knife)
+	living_user = user
+	RegisterSignal(user, COMSIG_MOB_MOUSEDRAG, .proc/change_target)
+	RegisterSignal(user, COMSIG_MOB_MOUSEUP, .proc/stop_fire)
+	RegisterSignal(user, COMSIG_MOB_MOUSEDOWN, .proc/start_fire)
 
 /obj/item/stack/throwing_knife/unequipped(mob/unequipper, slot)
 	. = ..()
+	living_user?.client?.mouse_pointer_icon = initial(living_user.client.mouse_pointer_icon) // Force resets the mouse pointer to default so it defaults when the last knife is thrown
 	UnregisterSignal(unequipper, COMSIG_MOB_ITEM_AFTERATTACK)
+	UnregisterSignal(unequipper, list(COMSIG_MOB_MOUSEUP, COMSIG_MOB_MOUSEDRAG, COMSIG_MOB_MOUSEDOWN))
+	living_user = null
+
+/obj/item/stack/throwing_knife/proc/change_target(datum/source, atom/src_object, atom/over_object, turf/src_location, turf/over_location, src_control, over_control, params)
+	SIGNAL_HANDLER
+	set_target(get_turf_on_clickcatcher(over_object, source, params))
+	living_user.face_atom(current_target)
+
+/obj/item/stack/throwing_knife/proc/stop_fire()
+	SIGNAL_HANDLER
+	living_user?.client?.mouse_pointer_icon = initial(living_user.client.mouse_pointer_icon)
+	SEND_SIGNAL(src, COMSIG_GUN_STOP_FIRE)
+
+/obj/item/stack/throwing_knife/proc/start_fire(datum/source, atom/object, turf/location, control, params)
+	SIGNAL_HANDLER
+	if(living_user.hand && !istype(living_user.l_hand, /obj/item/stack/throwing_knife/) || !living_user.hand && !istype(living_user.r_hand, /obj/item/stack/throwing_knife/)) // If the object in our active hand is not a throwing knife, abort
+		return
+	var/list/modifiers = params2list(params)
+	if(modifiers["shift"] || modifiers["ctrl"])
+		return
+	set_target(get_turf_on_clickcatcher(object, living_user, params))
+	if(!current_target)
+		return
+	SEND_SIGNAL(src, COMSIG_GUN_FIRE)
+	living_user?.client?.mouse_pointer_icon = 'icons/effects/supplypod_target.dmi'
 
 ///Throws a knife from the stack, or, if the stack is one, throws the stack.
-/obj/item/stack/throwing_knife/proc/throw_knife(datum/source, atom/target, params)
+/obj/item/stack/throwing_knife/proc/throw_knife()
 	SIGNAL_HANDLER
-	var/mob/living/user = source
-	if(user.Adjacent(target) || user.get_active_held_item() != src || !COOLDOWN_CHECK(src, last_thrown))
+	if(living_user.get_active_held_item() != src)
 		return
+	if(living_user.Adjacent(current_target))
+		return AUTOFIRE_CONTINUE
```

Comment:
?

Reply:
This is so that autofire continues when you drag your mouse near you. If this is removed, when you are holding click, if current_target becomes any tile adjacent to you, it stops firing. return AUTOFIRE_CONTINUE is so that when current_target leaves your adjacent it keeps firing and you dont have to click again.